login/src/main/java/com/fightpandemics/login/ui/LoginViewModel.kt

```diff
@@ -1,45 +1,90 @@
+import com.fightpandemics.core.data.model.login.CompleteProfileRequest
+import com.fightpandemics.core.data.model.login.CompleteProfileResponse
+import com.fightpandemics.core.data.model.login.LoginRequest
+import com.fightpandemics.core.data.model.login.LoginResponse
+import com.fightpandemics.core.data.model.login.SignUpRequest
+import com.fightpandemics.core.data.model.login.SignUpResponse
+import com.fightpandemics.core.data.model.login.User
+import com.fightpandemics.core.data.model.userlocation.LocationRequest
+import com.fightpandemics.core.data.model.userlocationpredictions.Prediction
 import com.fightpandemics.core.result.Result
 import com.fightpandemics.login.domain.CompleteProfileUseCase
+import com.fightpandemics.login.domain.LocationDetailsUseCase
+import com.fightpandemics.login.domain.LocationPredictionsNameAndIdUseCase
 import com.fightpandemics.login.domain.LoginUseCase
 import com.fightpandemics.login.domain.SignUPUseCase
+import com.fightpandemics.login.domain.UserLocationUseCase
 import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.Job
 import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.catch
 import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.conflate
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.launch
+import timber.log.Timber
 import javax.inject.Inject
 
+@FlowPreview
 @ExperimentalCoroutinesApi
 @ActivityScope
 class LoginViewModel @Inject constructor(
+    private val locationDetailsUseCase: LocationDetailsUseCase,
+    private val locationPredictionsNameAndIdUseCase: LocationPredictionsNameAndIdUseCase,
+    private val userLocationUseCase: UserLocationUseCase,
     private val loginUseCase: LoginUseCase,
     private val signUpUseCase: SignUPUseCase,
     private val completeProfileUseCase: CompleteProfileUseCase
 ) : ViewModel() {
+    // location variables
+    private var sharelocationJob: Job? = null
+    private var searchlocationJob: Job? = null
+    val searchQuery = MutableStateFlow("")
+
+    lateinit var completeProfileLocation: com.fightpandemics.core.data.model.login.Location
+
+    private val _currentLocationState = MutableStateFlow(UserLocationViewState(isLoading = true))
+    val currentLocationState = _currentLocationState.asStateFlow()
+    private val _searchLocationState = MutableStateFlow(mutableListOf<Prediction>())
+    val searchLocationState = _searchLocationState.asStateFlow()
+
+    // variable to keep track of selected location, also allows us to delete not-selected input from user
+    private val _locationSelected = MutableLiveData("")
+    val locationSelected = _locationSelected
```

Comment:
Should `locationSelected` and `_locationSelected` be equal? If so why are two different values needed?